custom_components/tuya_local/devices/poolex_arctic_action.yaml

```diff
@@ -0,0 +1,268 @@
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 21
+          value: "Ambient Temperature Fault"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 23
+          value: "Cooling Water Temperature Too Low Protection"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 27
+          value: "Outlet Water Temperature Fault"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 29
+          value: "Return Gas Temperature Fault"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 32
+          value: "Heating Water Temperature Too High Protection"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 33
+          value: "Outdoor Coil High Temperature Protection"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 35
+          value: "Compressor Current Protection"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+        - dps_val: 42
+          value: "Internal Coil Temperature Fault"
+          icon: "mdi:water-boiler-alert"
+          icon_priority: 2
+
+
+secondary_entities:
+  - entity: water_heater
+    icon: "mdi:hot-tub"
+    deprecated: climate
+    category: config
+    dps:
+      - id: 1
+        type: boolean
+        name: operation_mode
+        mapping:
+          - dps_val: false
+            value: "off"
+          - dps_val: true
+            value: heatpump
+      - id: 2
+        type: integer
+        name: temperature
+        range:
+          min: 5
+          max: 40
+        mapping:
+          - constraint: temperature_unit
+            conditions:
+              - dps_val: f
+                range:
+                  min: 41
+                  max: 104
+                value_redirect: temp_set_f
+      - id: 3
+        type: integer
+        name: current_temperature
+        mapping:
+          - constraint: temperature_unit
+            conditions:
+              - dps_val: f
+                value_redirect: temp_current_f
+      - id: 13
+        type: string
+        name: temperature_unit
+        mapping:
+          - dps_val: c
+            value: C
+          - dps_val: f
+            value: F
+      - id: 14
+        type: integer
+        name: temp_set_f
+        range:
+          min: 41
+          max: 104
+        hidden: true
+        optional: true
+      - id: 15
+        type: integer
+        name: temp_current_f
+        hidden: true
+        optional: true
+  - entity: select
+    name: Temperature unit
+    icon: "mdi:temperature-celsius"
+    category: config
+    dps:
+      - id: 13
+        type: string
+        name: option
+        mapping:
+          - dps_val: c
+            value: Celsius
+          - dps_val: f
+            value: Fahrenheit
+  - entity: binary_sensor
+    name: Status
```

Comment:
This should be an hvac_action attribute of the climate entity, with mapping to either heating or cooling  depending on the preset mode.

Reply:
But actually this is dp 1, which is just the power switch. There is no need for a separate running sensor, as you can tell from the state of the climate entity whether it is switched off, or to cooling or heating. And this shouldn't be hvac_action, as it isn't giving any more info than hvac_mode does.

Reply:
Changed the configuration, based on the feedback provided. 

Also fixed the product_id, but it is still not discovered after placing it in my local custom_components/tuya_local/devices/ folder. Only matches for 83% with hydrotherm_dynamic_x8_water_heater and it doesnt come up in the type selection dropdown. Don't know how to test properly I'm afraid.

```
{
  "result": [
    {
      "active_time": 1698766405,
      "bind_space_id": "170682830",
      "category": "qn",
      "create_time": 1698766405,
      "custom_name": "Poolex",
      "icon": "smart/icon/ay1513418289516FAXSV/cfca410c1da90b02660fc721b48a951b.png",
      "id": "_redacted_",
      "ip": "_redacted_",
      "is_online": true,
      "lat": "21.7300",
      "local_key": "_redacted_",
      "lon": "15.3100",
      "model": "NC11440004",
      "name": "Heat Pumps",
      "product_id": "pzp8owd7lhtpssgj",
      "product_name": "南晟变频泳池机 ℃ ℉",
      "sub": false,
      "time_zone": "+01:00",
      "update_time": 1698766527,
      "uuid": "5734565c743afbbf"
    }
  ],
  "success": true,
  "t": 1706715008831,
  "tid": "9dcec602c04d11ee84c772fd0bbfcb23"
}
```